### PR TITLE
Dropdown / Checkbox Group / Radiobutton list does not maintain order on UI

### PR DESCRIPTION
**Pull request by amanthakur03**

### CODE REVIEW

File: bundles/af-core/src/test/java/com/adobe/cq/forms/core/components/internal/models/v1/form/DropDownImplTest.java [hunk 1]
```diff
@@ -15,9 +15,7 @@
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
 package com.adobe.cq.forms.core.components.internal.models.v1.form;
 
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.Map;
```
Comment (rismehta) [inline bundles/af-core/src/test/java/com/adobe/cq/forms/core/components/internal/models/v1/form/DropDownImplTest.java:20]: Please fix the imports

File: bundles/af-core/src/test/java/com/adobe/cq/forms/core/components/internal/models/v1/form/RadioButtonImplTest.java [hunk 2]
```diff
@@ -15,9 +15,7 @@
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
 package com.adobe.cq.forms.core.components.internal.models.v1.form;
 
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.Map;
+import java.util.*;
```
Comment (rismehta) [inline bundles/af-core/src/test/java/com/adobe/cq/forms/core/components/internal/models/v1/form/RadioButtonImplTest.java:18]: Please fix the imports everywhere

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractOptionsFieldImpl.java [hunk 3]
```diff
@@ -64,16 +61,16 @@ public boolean isEnforceEnum() {
      *         {@code @else}
      * @return map containing enum values and enum names as key-value pairs
      */
-    private Map<Object, String> removeDuplicates() {
+    private LinkedHashMap<Object, String> removeDuplicates() {
```
Comment (barshat7) [inline bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractOptionsFieldImpl.java:64]: Ideally you should return/accept an interface. No need to change the signature.

**Comment by barshat7** [comment]
LGTM

**Review by barshat7 — APPROVED** [review]

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractOptionsFieldImpl.java [hunk 4]
```diff
@@ -67,13 +67,12 @@ public boolean isEnforceEnum() {
     private Map<Object, String> removeDuplicates() {
 
         Object[] enumArray = this.enums;
+        Preconditions.checkNotNull(enumArray, "Variable must not be null");
```
[the remark this answers is not in the document]
Comment (rismehta) [reply bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractOptionsFieldImpl.java:70]: Also, I prefer null checks makes the code more readable and self-documenting by clearly expressing the intent that a certain value should not be null.

**Review by rismehta — CHANGES_REQUESTED** [review]
-1 for now

**Pull request merged**